caddyhttp/proxy/reverseproxy.go

```diff
@@ -122,10 +122,44 @@ func NewSingleHostReverseProxy(target *url.URL, without string) *ReverseProxy {
 		rp.Transport = &http.Transport{
 			Dial: socketDial(target.String()),
 		}
+	} else if keepalive != 0 {
+		rp.Transport = &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		}
+		if keepalive < 0 {
+			rp.Transport.(*http.Transport).DisableKeepAlives = true
+		} else {
+			rp.Transport.(*http.Transport).MaxIdleConnsPerHost = keepalive
+		}
```

Comment:
Hmm. I think either way is fine, the more I think about it. If we hard-code a default (like 2) and then _if_ people report problems in the future due to some change, then I guess we can just update Caddy's default. It wouldn't be any harder than recompiling with a new version of Go, since a new compile would have to happen anyway.

I might regret this later but let's change it so that 0 disables keepalives; nginx apparently doesn't enable them by default but Caddy should, so there should be an easy way to turn them off.